File: 프로그래머스/3/42884. 단속카메라/단속카메라.py
def solution(routes):
    routes.sort(key=lambda x: (x[1], x[0]))
    print(routes)
    now = routes[0][1]
    answer = 1
    for route in routes:
        if route[0] <= now:
            continue
        now = route[1]
        answer += 1
    return answer

File: 프로그래머스/3/42884. 단속카메라/단속카메라.java
import java.util.*;

class Solution {
    public int solution(int[][] routes) {
        Arrays.sort(routes, (a, b) -> {
            if (a[1] == b[1]) {
                return a[0]-b[0];
            }
            return a[1]-b[1];
        });
        
        int now = routes[0][1];
        int answer = 1;
        
        for (int i=0; i<routes.length; i++) {
            if (routes[i][0] <= now) {
                continue;
            }
            now = routes[i][1];
            answer++;
        }
        
        return answer;
    }
}